jsonrpc/http_test.go

```diff
@@ -74,3 +78,107 @@ func TestHTTP(t *testing.T) {
 		assert.Len(t, listener.OnNewRequestLogs, 1)
 	})
 }
+
+func TestGzipResponse(t *testing.T) {
+	method := jsonrpc.Method{
+		Name: "echo",
+		Handler: func(msg string) (string, *jsonrpc.Error) {
+			return msg, nil
+		},
+		Params: []jsonrpc.Parameter{{Name: "msg"}},
+	}
+	listener := CountingEventListener{}
+	log := utils.NewNopZapLogger()
+	rpc := jsonrpc.NewServer(1, log).WithListener(&listener)
```

Comment:
If we don't use the listener:
```suggestion
	rpc := jsonrpc.NewServer(1, log)
```